tp1/tp1.go

```diff
@@ -24,17 +49,98 @@ type Carrito struct {
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+
+	mapaCarritos := map[string]int{}
+
+	for _, id := range ids {
+		for _, producto := range p {
+
+			if producto[POSICION_ID] == strconv.Itoa(id) {
+				mapaCarritos[producto[POSICION_TIENDA]] =
+					mapaCarritos[producto[POSICION_TIENDA]] + parseInt(producto[POSICION_PRECIO])
+			}
+		}
+	}
+
+	var carritos []Carrito
+
+	for tienda, precio := range mapaCarritos {
+		carritos = append(carritos, Carrito{Tienda: tienda, Precio: precio})
+	}
+
+	return carritos
 }
 
 // Promedio recibe el id de un producto y retorna el precio promedio
 // de ese producto usando los precios de todos los supermercados.
 func (p Productos) Promedio(idProducto int) float64 {
-	return 0
+
+	tiendas := p.buscarPorId(idProducto)
+
+	cantidad := float64(len(tiendas))
+	sumaTotal := 0.0
+	var promedio float64
+
+	if cantidad > 0 {
+		for _, precioStr := range tiendas {
+			sumaTotal = sumaTotal + parseFloat(precioStr)
+		}
+
+		promedio = sumaTotal / cantidad
+	}
+
+	return promedio
 }
 
 // BuscarMasBarato recibe un id de producto a buscar y te retorna
 // el producto mas barato que haya encontrado.
 func (p Productos) BuscarMasBarato(idProducto int) (Producto, bool) {
-	return nil, false
+
+	tiendas := p.buscarPorId(idProducto)
+
+	productoMasBarato := producto{id: idProducto}
+	var encontrado bool
+
+	if tiendas != nil && len(tiendas) > 0 {
```

Comment:
Aca tambien aplicaria el sad path first. Busquemos la condicion que nos haga retornar primero:
```
if tiendas == nil || len(tiendas) == 0 {
    return producto{}, false
}
```

Reply:
De esta manera reducimos tambien un nivel de indentacion y mantenemos el codigo que hace lo que queremos mas cerca

Reply:
Esta charla de mat ryer: https://www.youtube.com/watch?v=yeetIgNeIkc tiene una buena explicacion de esto. Y lo llama "line of sight", que basicamente es mantener el codigo que hace lo que queremos lo menos indentado posible

Reply:
Dale, lo voy a tener en cuenta, gracias